### PR TITLE
test: increasing unit tests coverage

### PR DESCRIPTION
**Pull request by kosegor**
## Description

closes: #484

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added `!` to the type prefix if API or client breaking change
- [x] added appropriate labels to the PR
- [x] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [x] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [x] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/uibc/quota/keeper/quota_test.go [hunk 1]
```diff
@@ -25,3 +41,117 @@ func TestResetQuota(t *testing.T) {
...
+	leverageMlk.EXPECT().GetTokenSettings(ctx, "atom").Return(
+		lfixtures.Token("atom", "ATOM", 6), nil,
+	).AnyTimes()
+	oracleMlk.EXPECT().Price(ctx, "ATOM").Return(sdk.Dec{}, types.ErrMalformedLatestAvgPrice)
+
+	err = k.CheckAndUpdateQuota(ctx, atomToken.Denom, atomToken.Amount)
+	assert.ErrorIs(t, err, types.ErrMalformedLatestAvgPrice)
+
+	// dai
+	leverageMlk.EXPECT().GetTokenSettings(ctx, "dai").Return(
```
Comment (toteki) [inline x/uibc/quota/keeper/quota_test.go:92]:
```suggestion
	// DAI returns token settings and prices from mock leverage and oracle keepers
	leverageMlk.EXPECT().GetTokenSettings(ctx, "dai").Return(

```
Increase comments when writing long test scenarios

File: x/uibc/quota/keeper/quota_test.go [hunk 2]
```diff
@@ -25,3 +41,117 @@ func TestResetQuota(t *testing.T) {
...
+		lfixtures.Token("dai", "DAI", 6), nil,
+	).AnyTimes()
+	oracleMlk.EXPECT().Price(ctx, "DAI").Return(sdk.MustNewDecFromStr("0.37"), nil)
+
+	err = k.SetParams(ctx, uibc.DefaultParams())
+	assert.NilError(t, err)
```
Comment (toteki) [inline x/uibc/quota/keeper/quota_test.go:98]:
```suggestion
	err = k.SetParams(ctx, uibc.DefaultParams())
	assert.NilError(t, err)
```

Were params changed before they were set to DefaultParams in this scenario? A comment would clarify whether this is performing a param change required for the test scenario or just testing `SetParams`.

File: x/uibc/quota/keeper/quota_test.go [hunk 3]
```diff
@@ -25,3 +41,117 @@ func TestResetQuota(t *testing.T) {
...
+	k.SetTokenOutflows(ctx, setQuotas)
+
+	err = k.UndoUpdateQuota(ctx, umeeToken.Denom, umeeToken.Amount)
+	assert.NilError(t, err)
+
+	o, err = k.GetTokenOutflows(ctx, umeeToken.Denom)
+	assert.NilError(t, err)
+
+	powerReduction := sdk.MustNewDecFromStr("10").Power(uint64(umeeExponent))
+	expectedQuota := sdk.NewDec(umeeQuota.Int64()).Sub(sdk.NewDecFromInt(umeeToken.Amount).Quo(powerReduction).Mul(umeePrice))
```
Comment (toteki) [inline x/uibc/quota/keeper/quota_test.go:155]: comments in this test scenario explaining expected values would make it more clear

**Review by toteki — APPROVED** [review]
Good work - I left a couple comments but it should be good to merge overall

File: x/leverage/types/genesis_test.go [hunk 4]
```diff
@@ -3,12 +3,155 @@ package types
...
+		},
+		{
+			"invalid params", GenesisState{
+				Params: Params{
+					CompleteLiquidationThreshold: sdk.MustNewDecFromStr("-0.4"),
+				},
+			},
+			true,
+			"complete liquidation threshold must be positive",
+		},
```
[the remark this answers is not in the document]
Comment (kosegor) [reply x/leverage/types/genesis_test.go:40]: yep, good catch

File: x/leverage/types/params_test.go [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+		err  string
+	}{
+		{"default params", DefaultParams(), ""},
+		{
+			"negative complete liquidation threshold",
+			Params{
+				CompleteLiquidationThreshold: negativeDec,
+			},
+			"complete liquidation threshold must be positive",
+		},
```
[the remark this answers is not in the document]
Comment (kosegor) [reply x/leverage/types/params_test.go:29]: added

File: x/uibc/quota/keeper/quota_test.go [hunk 6]
```diff
@@ -25,3 +41,117 @@ func TestResetQuota(t *testing.T) {
...
+	defer leverageCtrl.Finish()
+	leverageMlk := mocks.NewMockLeverageKeeper(leverageCtrl)
+
+	oracleCtrl := gomock.NewController(t)
+	defer oracleCtrl.Finish()
+	oracleMlk := mocks.NewMockOracle(oracleCtrl)
+
+	interfaceRegistry := ctypes.NewInterfaceRegistry()
+	marshaller := codec.NewProtoCodec(interfaceRegistry)
+	ctx, k := initFullKeeper(t, marshaller, nil, leverageMlk, oracleMlk)
```
[the remark this answers is not in the document]
Comment (kosegor) [reply x/uibc/quota/keeper/quota_test.go:61]: done

File: x/uibc/quota/keeper/quota_test.go (hunk 1 above)
Comment (kosegor) [reply x/uibc/quota/keeper/quota_test.go:92]: added more comment to the tests

File: x/uibc/quota/keeper/quota_test.go (hunk 3 above)
Comment (kosegor) [reply x/uibc/quota/keeper/quota_test.go:157]: done

**Pull request merged**